src/EditorFeatures/Core/AutomaticCompletion/BraceCompletionSessionProvider.BraceCompletionSession.cs

```diff
@@ -193,52 +198,48 @@ public void PreOverType(out bool handledCommand)
+                }
 
-                    // ensure that we are within the session before clearing
-                    if (caretPos.HasValue && caretPos.Value.Position < closingSnapshotPoint.Position && closingSnapshotPoint.Position > 0)
-                    {
-                        using var undo = CreateUndoTransaction();
+                var caretPos = this.GetCaretPosition();
 
-                        _editorOperations.AddBeforeTextBufferChangePrimitive();
+                Debug.Assert(caretPos.HasValue && caretPos.Value.Position < closingSnapshotPoint.Position);
 
-                        var span = new SnapshotSpan(caretPos.Value, closingSnapshotPoint.Subtract(1));
+                // ensure that we are within the session before clearing
+                if (caretPos.HasValue && caretPos.Value.Position < closingSnapshotPoint.Position && closingSnapshotPoint.Position > 0)
+                {
+                    using var undo = CreateUndoTransaction();
 
-                        using var edit = SubjectBuffer.CreateEdit();
+                    _editorOperations.AddBeforeTextBufferChangePrimitive();
 
-                        edit.Delete(span);
+                    var span = new SnapshotSpan(caretPos.Value, closingSnapshotPoint.Subtract(1));
 
-                        if (edit.HasFailedChanges)
-                        {
-                            Debug.Fail("Unable to clear closing brace");
-                            edit.Cancel();
-                            undo.Cancel();
-                        }
-                        else
-                        {
-                            handledCommand = true;
+                    using var edit = SubjectBuffer.CreateEdit();
 
-                            edit.ApplyAndLogExceptions();
+                    edit.Delete(span);
 
-                            MoveCaretToClosingPoint();
+                    if (edit.HasFailedChanges)
+                    {
+                        Debug.Fail("Unable to clear closing brace");
+                        edit.Cancel();
+                        undo.Cancel();
+                    }
+                    else
+                    {
+                        handledCommand = true;
 
-                            _editorOperations.AddAfterTextBufferChangePrimitive();
+                        edit.ApplyAndLogExceptions();
 
-                            undo.Complete();
-                        }
-                    }
-                }
+                        MoveCaretToClosingPoint();
 
-                return;
+                        _editorOperations.AddAfterTextBufferChangePrimitive();
 
-                bool AllowOverType()
-                {
-                    var context = GetBraceCompletionContext();
-                    return context != null && _service.AllowOverTypeAsync(context.Value, cancellationToken).WaitAndGetResult(cancellationToken);
+                        undo.Complete();
+                    }
                 }
             }
```

Comment:
not needed anymore?

Reply:
I inlined this and unindented 2 levels of indentation above.